lib/private/User/FailedUsersBackend.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+/**
+ * SPDX-FileCopyrightText: 2024 Nextcloud GmbH and Nextcloud contributors
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+namespace OC\User;
+
+use OCP\IConfig;
+use OCP\IUserBackend;
+use OCP\User\Backend\IGetHomeBackend;
+
+/**
+ * This is a "fake" backend for users that were deleted,
+ * but not properly removed from Nextcloud (e.g. an exception occurred).
+ * This backend is only needed because some APIs in user-deleted-events require a "real" user with backend.
+ */
+class FailedUsersBackend extends Backend implements IGetHomeBackend, IUserBackend {
```

Comment:
i think the naming gives a wrong and/or unclear impression.

```suggestion
class PartiallyDeletedUsersBackend extends Backend implements IGetHomeBackend, IUserBackend {
```

maybe?

Reply:
sounds better :sweat_smile: